libs/fn/src/lib/input/input-base.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * This abstract class contains common functionality that will be used by many different input components, such as the input, text area,
+ * select, multi-input etc.
+ */
+
+import { Directive, Input } from '@angular/core';
+
+export type InputState = 'positive' | 'critical' | 'negative' | 'info';
+
+@Directive()
+export abstract class InputBase {
+    /** Placeholder for the input. */
+    @Input()
+    placeholder: string;
+
+    /** Whether or not this input is the 'display' type. */
+    @Input()
+    display = false;
+
+    /** Whether or not this input is disabled. */
+    @Input()
+    fnDisabled = false;
```

Comment:
This & `fnReadonly` input properties redundant as for me. 

There is no coercion but it's passed as is to the appropriate directives, in case these fields will be used before coercion we may face unexpected behavior.

Also, appropriate directives can be used theirselves and we can get them (to pass further or anything) in the construction, of course with the `@Optional()`.

cc @g-cheishvili @N1XUS.

Reply:
Good call, see my recent commits for the update